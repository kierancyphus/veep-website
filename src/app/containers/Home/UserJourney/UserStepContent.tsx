import React, { FC, useState } from 'react'
import { WithActiveStep, WithSetStep } from 'types/home'
import {
  Box,
  Card,
  CardContent,
  IconButton,
  Typography,
  Button,
  Link,
} from '@material-ui/core'
import { ChevronLeft, ChevronRight } from 'react-feather'
import { Fade } from 'react-reveal'
import { Link as RouterLink } from 'react-router-dom'
import { WithUserType } from 'types/user'
import { WithProjectType } from 'types/projects'

export interface UserStepContentProps
  extends WithSetStep,
    WithActiveStep,
    WithProjectType,
    WithUserType {
  dates: string
  content: string
  length: number
  links?: Record<string, string>
}

enum Direction {
  NONE,
  LEFT,
  RIGHT,
}

export const UserStepContent: FC<UserStepContentProps> = ({
  activeStep,
  setStep,
  dates,
  content,
  links,
  length,
  projectType,
  userType,
}) => {
  const [direction, setDirection] = useState<Direction>(Direction.NONE)
  return (
    <Box display="flex" justifyContent="center">
      <Card elevation={5}>
        <Box display="flex" justifyContent="space-between">
          <IconButton
            disabled={activeStep === 0}
            onClick={() => {
              setStep(step => step - 1)
              setDirection(Direction.LEFT)
            }}
          >
            <ChevronLeft size={40} />
          </IconButton>
          <Fade
            left={direction === Direction.LEFT}
            right={direction === Direction.RIGHT}
            // key={`step-content-${activeStep}-${direction}`}
          >
            <CardContent>
              <Typography variant="h6">{`Important Dates: ${dates}`}</Typography>
              <br />
              <Typography variant="body1">{content}</Typography>
              {links && (
                <>
                  <br />
                  <Box display="flex" justifyContent="center">
                    {Object.entries(links).map(([text, href]) => (
                      <Link
                        component={RouterLink}
                        to={`${href}?projectType=${projectType}&userType=${userType}`}
                        key={`${text}-${href}`}
                      >
                        <Button key={`${text}-${href}`}>{text}</Button>
                      </Link>
                    ))}
                  </Box>
                </>
              )}
            </CardContent>
          </Fade>

          <IconButton
            disabled={activeStep === length}
            onClick={() => {
              setStep(step => step + 1)
              setDirection(Direction.RIGHT)
            }}
          >
            <ChevronRight size={40} />
          </IconButton>
        </Box>
      </Card>
    </Box>
  )
}
