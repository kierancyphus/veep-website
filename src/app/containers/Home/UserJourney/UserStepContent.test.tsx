import React from 'react'
import { UserStepContent, UserStepContentProps } from './UserStepContent'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

describe('UserStepContent', () => {
  const props: UserStepContentProps = {
    dates: 'January',
    content: 'this is some content',
    length: 3,
    activeStep: 0,
    setStep: jest.fn(),
    projectType: 0,
    userType: 0,
  }

  it('renders without errors', () => {
    const { container: firstChild } = render(<UserStepContent {...props} />, {
      wrapper: BrowserRouter,
    })

    expect(firstChild).toMatchSnapshot()
  })

  // figure out how to find the chevron

  // it('increments step on right chevron click', () => {
  //   const { getByText } = render(<UserStepContent {...props} />, {
  //     wrapper: Providers,
  //   })
  // })

  // it('decrements step on right chevron click', () => {
  //   const { getByText } = render(<UserStepContent {...props} />, {
  //     wrapper: Providers,
  //   })
  // })

  // it('disables left chevron on first step', () => {
  //   const { getByText } = render(<UserStepContent {...props} />, {
  //     wrapper: Providers,
  //   })
  // })

  // it('disables right chevron on right step', () => {
  //   const { getByText } = render(<UserStepContent {...props} />, {
  //     wrapper: Providers,
  //   })
  // })
})
