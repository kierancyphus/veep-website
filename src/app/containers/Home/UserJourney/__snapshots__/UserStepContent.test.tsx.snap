// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStepContent renders without errors 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation5 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              fill="none"
              height="40"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="15 18 9 12 15 6"
              />
            </svg>
          </span>
        </button>
        <div
          class="react-reveal"
        >
          <div
            class="MuiCardContent-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              Important Dates: January
            </h6>
            <br />
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              this is some content
            </p>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              fill="none"
              height="40"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
